src/transformer/utils/get_activity/user_report.php

```diff
@@ -19,20 +19,26 @@
 
 use src\transformer\utils as utils;
 
-function user_report(array $config, \stdClass $user, \stdClass $course) {
-    $courselang = utils\get_course_lang($course);
+function user_report(array $config, \stdClass $user, \stdClass $course, $relateduserid, $courselang) {
 
-    return [
-        'id' => $config['app_url'].'/mod/forum/user.php?id='.$user->id.'&course='.$course->id,
+    $activity = [
         'definition' => [
             'type' => 'http://id.tincanapi.com/activitytype/user-profile',
             'name' => [
                 $courselang => 'forum posts of '.utils\get_full_name($user),
```

Comment:
Since we're using the related user ID below for the activity ID, does that mean we should use the related user here?

Reply:
@garemoko

Reply:
Yes, definitely. Well spotted!

Reply:
In which case i probably shouldn't have modified the user part of this file, but instead should have just changed the user that is passed in by user_report_viewed.php

Reply:
Sweet. Yeah good idea that probably makes the most sense.

Reply:
@ryansmith94 this is resolved.